packages/mask/src/components/InjectedComponents/SetupGuide.tsx

```diff
@@ -41,186 +35,112 @@ interface SetupGuideUIProps {
-    const platform = ui.configuration.nextIDConfig?.platform as NextIDPlatform
-    // #region parse setup status
-    const lastState_ = useValueRef(currentSetupGuideStatus[ui.networkIdentifier])
-    const lastState = useMemo<SetupGuideContext>(() => {
-        try {
-            return JSON.parse(lastState_)
-        } catch {
-            return {}
-        }
-    }, [lastState_])
-    useEffect(() => {
-        setStep(lastState.status ?? SetupGuideStep.Close)
-    }, [lastState])
-    // #endregion
 
-    // #region setup username
-    const lastRecognized = useLastRecognizedIdentity()
-    const getUsername = () => lastState.username || lastRecognized.identifier?.userId || ''
-    const [username, setUsername] = useState(getUsername)
+    const {
+        value: { step, userId, currentIdentityResolved, destinedPersonaInfo, type } = {
+            step: SetupGuideStep.Close,
+            userId: '',
+            type: 'close',
+        },
+        loading: loadingSetupGuideStep,
+        retry: retryCheckStep,
+    } = useSetupGuideStepInfo(persona)
 
-    const disableVerify =
-        !lastRecognized.identifier || !lastState.username
-            ? false
-            : lastRecognized.identifier.userId !== lastState.username
+    // #region should not show notification if user have operation
+    const [hasOperation, setOperation] = useState(false)
 
     useEffect(() => {
-        const handler = (val: IdentityResolved) => {
-            if (username === '' && val.identifier) setUsername(val.identifier.userId)
-        }
-        return ui.collecting.identityProvider?.recognized.addListener(handler)
-    }, [username])
+        if (!(type === 'done' && !hasOperation)) return
+        showSnackbar(t('setup_guide_connected_title'), {
+            variant: 'success',
+            message: t('setup_guide_connected_description'),
+        })
 
-    useEffect(() => {
-        if (username || ui.networkIdentifier !== EnhanceableSite.Twitter) return
-        // In order to collect user info after login, need to reload twitter once
-        let reloaded = false
-        const handler = () => {
-            // twitter will redirect to home page after login
-            if (!(!reloaded && location.pathname === '/home')) return
-            reloaded = true
-            location.reload()
-        }
-        window.addEventListener('locationchange', handler)
-        return () => {
-            window.removeEventListener('locationchange', handler)
-        }
-    }, [username])
+        /**
+         * why use setTimeout?
+         *
+         * In this case: User opened the twitter.com page, and then click `connect social media` from Popup
+         * The popup will create new tab to open the twitter.com.
+         * In old twitter.com, the follow setter will execute immediately,
+         * and the new twitter.com page need time to load, and will get the empty `currentSetupGuideStatus`.
+         */
+        setTimeout(() => {
```

Comment:
We need to clean this timer.

Reply:
I create a new pr try to fix this issue: https://github.com/DimensionDev/Maskbook/pull/7387/files